database/src/adapters/execution_outcomes.rs

```diff
@@ -18,7 +18,7 @@ pub async fn store_execution_outcomes(
             &shard.receipt_execution_outcomes,
             shard.shard_id,
             block_timestamp,
-            std::sync::Arc::clone(&receipts_cache),
+            receipts_cache.clone(),
```

Comment:
The reason behind using "scary" `std::sync::Arc::clone()` was explicitness. And I really don't want to get rid of it.

With the "scary" call a reader understands that it's an `Arc` clone at a glance. With your change, a reader wouldn't be able to get it immediately without diving into the codebase.

Please, revert this change.

Reply:
I disagree.
In master, we already have 3 places where we use `receipts_cache.clone()`: it's just impossible to remember that we have to use an explicit form

If you go to the definition of ReceiptsCache, you see it's Arc, it's not diving deep in my opinion.

And, btw, why it's important for the reader to know it's an Arc clone?

Reply:
> In master, we already have 3 places where we use receipts_cache.clone(): it's just impossible to remember that we have to use an explicit form

That highlights someone's mistakes (most likely mine). I would appreciate it if you update them to explicit calls as well :)

> If you go to the definition of ReceiptsCache, you see it's Arc, it's not diving deep in my opinion.

Can't argue with that it would be obvious if a reader had seen it.

> And, btw, why it's important for the reader to know it's an Arc clone?

I don't know, please accept it as my personal psychological issues. The project is somehow complicated, we are transferring the maintenance to another person and I would prefer we keep caring about the possible readers.

Reply:
As a reader and non-native Rust speaker, when I see `std::sync::Arc::clone`, I think "wow, I'd never thought I have to use so unintuitive form, I'll never write valid Rust code"
Moreover, while developing Indexer for the last 2 years, I've read the definition of `std::sync::Arc::clone` several times just because it disrupts me from reading, and I need to check why we have to use this. The truth is: we don't have to, this logic is going out of the box.

The only situation when you need to know that it's an Arc clone and the strong pointers will be created under the hood, is the moment when you've already dived super deep. In this case, the developer will not need this code, they are already much deeper than we are.

Reply:
I think each of your points are valid:
1. Why be explicit when `Arc` provides the `.clone()` syntax/functionality out of the box
2. It's not immediately obvious that this is an `Arc`, within an IDE it's not too hard to figure out, but it's still mental overhead that is better avoided.

How about this for a middle ground solution:
* Rename `ReceiptsCache` type -> `ReceiptsCacheArc`
* Rename all `receipt_cache` variables -> `receipts_cache_arc`

This way the explicitness is tied to the variable/type itself and we don't need to remember to use `std::sync::Arc::clone` everywhere, instead we can just use `.clone()`

Thoughts? :)

Reply:
I like this!
@khorolets what do you think?

Reply:
Yeah I can live with that. Thanks @morgsmccauley